src/traversals.js

```diff
@@ -1,9 +1,17 @@
+import * as F from 'futil-js'
 import { Tree } from './util/tree'
 
 export default extend => ({
   markForUpdate: Tree.walk(x => {
     if (x.paused) extend(x, { missedUpdate: true })
-    else extend(x, { markedForUpdate: true })
+    else {
+      let updatingDeferred = F.defer()
```

Comment:
Why use `defer` here?

Reply:
the use of `defer` is to resolve the promise externally when updating finishes